bands/site_sections/bands.py

```diff
@@ -9,143 +9,172 @@ def index(self, session, id, message=''):
             'band': session.band(id)
         }
 
-    def agreement(self, session, message='', **params):
-        band = session.band(params, restricted=True)
+    def agreement(self, session, band_id, id=None, message='', **params):
+        band = session.band(band_id)
+        band_info = session.band_info(params, restricted=True)
         if cherrypy.request.method == 'POST':
-            if not band.performer_count:
+            if not band_info.performer_count:
                 message = 'You must tell us how many people are in your band'
-            elif not band.poc_phone:
+            elif not band_info.poc_phone:
                 message = 'You must enter an on-site point-of-contact cellphone number'
-            elif not band.arrival_time:
+            elif not band_info.arrival_time:
                 message = 'You must enter your expected arrival time'
-            elif band.bringing_vehicle and not band.vehicle_info:
+            elif band_info.bringing_vehicle and not band_info.vehicle_info:
                 message = 'You must provide your vehicle information'
             else:
+                band.info = band_info
+                session.merge(band_info)
```

Comment:
Either one is fine here.  FWIW, calling ``session.merge(band_info)`` actually detaches the ``band_info`` object you pass it from the session and then returns a new ``BandInfo`` instance which is now attached to the session.  I usually avoid calling it unless that's explicitly what I want to happen.  In this case it would cause problems if you tried to use the original ``band_info`` after calling ``session.merge`` on it, but since you aren't it's fine here.  In general though, I'd tend to use ``session.add()`` unless I really want that detach-and-copy behavior.